ArcTouchTMDb/ArcTouchTMDb.Core/Services/API/RestAPIService.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Diagnostics;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using ArcTouchTMDb.Core.Services.API.Request;
+using ArcTouchTMDb.Core.Services.API.Response;
+using Newtonsoft.Json;
+
+namespace ArcTouchTMDb.Core
+{
+	public abstract class RestAPIService
+	{
+		private HttpClient client;
+
+		public RestAPIService()
+		{
+			client = new HttpClient();
+		}
+
+		public async Task<T> Request<T>(IBaseRequest request) where T : IBaseResponse
+		{
+			try
+			{
+				var response = await client.GetAsync(request.CreateRequestUrl());
+
+				if (response.IsSuccessStatusCode)
```

Comment:
Good job checking the status code, but I think we could have a way to handle other statuses here: simply an `else` that throws an exception that can be handled by the calling method.